src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.File.Base.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Formats.Tar.Tests
+{
+    public partial class TarWriter_File_Base : TarTestsBase
+    {
+        // TestTarFormat, string testCaseName, CompressionMethod, bool copyData
+        public static IEnumerable<object[]> WriteEntry_CopyArchive_Data()
+        {
+            foreach (CompressionMethod compressionMethod in Enum.GetValues<CompressionMethod>())
+            {
+                foreach (bool copyData in new object[] { false, true })
+                {
+                    foreach (object[] testCaseName in GetV7TestCaseNames())
+                    {
+                        if ((string)testCaseName[0] == "folder_file_utf8") // The legacy name field is stored in ASCII in the V7 format
+                        {
+                            continue;
+                        }
+
+                        yield return new object[] { TestTarFormat.v7, testCaseName[0], compressionMethod, copyData };
+                    }
+
+                    foreach (object[] testCaseNameObj in GetUstarTestCaseNames())
+                    {
+                        string testcaseName = (string)testCaseNameObj[0];
+                        if (testcaseName is "folder_file_utf8" // The legacy name field is stored in ASCII in the Ustar format
+                                         or "longpath_splitable_under255" // Folder stored under 'unarchived' runtime-assets due to NuGet not allowing very long paths
+                                         or "specialfiles") // Same but for fifos/chardevices/blockdevices
+                        {
+                            continue;
+                        }
+
+                        yield return new object[] { TestTarFormat.ustar, testcaseName, compressionMethod, copyData };
+                    }
+
+                    foreach (object[] testCaseNameObj in GetPaxAndGnuTestCaseNames())
+                    {
+                        string testCaseName = (string)testCaseNameObj[0];
+                        if (testCaseName is "longpath_splitable_under255"
+                                         or "specialfiles"
+                                         or "longpath_over255" // Folder stored under 'unarchived' runtime-assets due to NuGet not allowing very long paths
+                                         or "longfilename_over100_under255" // Same
+                                         or "file_longsymlink") // Same
```

Comment:
Do you mind elaborate further why are you skipping these archives? and how do you deal with them in other tests?

Reply:
For the first two, the reason is the same as the one I wrote a few lines above.

Generally speaking, these tests are collecting the unarchived files from the filesystem. We have to skip them because either some OS variants do not support the files in these test cases, or nuget is unable to pack them from runtime-assets and unpack them unchanged into the test data.

Other tests are not consuming the unarchived files, they are consuming the .tar or .tar.gz files directly, which we can iterate without interacting with the filesystem. But this test in particular needs to be able to compare with the filesystem, not with TarReader entries.

The skipped test cases aren't that important. What's important is the copy of an entry that comes from a TarReader into a new TarWriter, particularly in the PAX format, to verify the bug is fixed in the extended attributes.

Reply:
> this test in particular needs to be able to compare with the filesystem, not with TarReader entries.

I left a suggestion below that can help you avoid comparing against the filesystem https://github.com/dotnet/runtime/pull/75237#discussion_r966353141. Could that help removing all these `continue`s?